packages/clerk-js/src/utils/url.ts

```diff
@@ -352,20 +353,25 @@ export function requiresUserInput(redirectUrl: string): boolean {
   return frontendApiRedirectPathsWithUserInput.includes(url.pathname);
 }
 
-export const isAllowedRedirectOrigin =
-  (allowedRedirectOrigins: Array<string | RegExp> | undefined) => (_url: string) => {
-    if (!allowedRedirectOrigins) {
-      return true;
+export const isAllowedRedirect =
```

Comment:
This function now deals with full URLs, instead of potentially relative URLs as well. This makes our checks more consistent, and removes situations where we treat a relative URL as okay, make it absolute, and then it ends up being problematic.